### ISSUE TITLE
Generate code for override functions from virtual functions

### ISSUE THREAD

**Issue by timxian** [microsoft/vscode-cpptools#4737]
It would be very useful if cpptools could have some auto-generation features:
1.  generate methods functions from header file to source file
2.  generate override functions from virtual functions of current class or parent class

**Comment by sean-mcmanus** [comment]
We're tracking issue 1 with https://github.com/microsoft/vscode-cpptools/issues/664 .

We can track issue 2 with this issue.

**Comment by timxian** [comment]
> We're tracking issue 1 with #664 .
> 
> We can track issue 2 with this issue.

Great! Looking forward to have this feature

**Comment by drinktee** [comment]
Are there further developments in the feature?

**Comment by sean-mcmanus** [comment]
No, but we're working on https://github.com/microsoft/vscode-cpptools/issues/664

**Comment by Raildex** [comment]
more than 2 years now. Is there any development going on in the first place?

**Comment by bobbrow** [comment]
Yes, there is development going on.  You can follow the releases here: https://github.com/microsoft/vscode-cpptools/releases.  You can also see what we're currently working on here: https://github.com/microsoft/vscode-cpptools/projects?type=classic

**Comment by sean-mcmanus** [comment]
@Raildex As mentioned previously, the 1st issue is being actively developed. Also, the 2nd issue seems like it would share a lot of the same implementation as the 1st issue, so it might be relatively straightforward to add after the 1st issue is addressed. Both features are currently available in VS (on Windows).

**Comment by beogivaljr** [comment]
Any improvement on the "generate override functions from virtual functions of current class or parent class" issue?

**Comment by timxian** [comment]
> @Raildex As mentioned previously, the 1st issue is being actively developed. Also, the 2nd issue seems like it would share a lot of the same implementation as the 1st issue, so it might be relatively straightforward to add after the 1st issue is addressed. Both features are currently available in VS (on Windows).

@sean-mcmanus sice 1st issue has been release, is there any plan to work on 2nd issue ?

**Comment by sean-mcmanus** [comment]
@timxian Not sure yet. I'm looking into it.

**Comment by sean-mcmanus** [comment]
Okay, I see it implemented in VS --  I don't know when this will get on our schedule yet, i.e. we have other high priority features on deck before this.

**Comment by tadmn** [comment]
Any update on the 2nd issue? Looking forward to this.

**Comment by Dragon-qing** [comment]
Looking forward to 2nd issue..
